SampleService.spec

```diff
@@ -499,4 +499,17 @@ module SampleService {
        for the service.
      */
     funcdef get_data_link(GetDataLinkParams params) returns(DataLink link) authentication required;
+
+    /* Provide sample and run through the validation steps, but without saving them. Allows all the samples to be evaluated for validity first so potential errors can be addressed.
+    */
+
+    typedef structure {
+        list<Sample> samples;
+    } ValidateSamplesParams;
+
+    typedef structure {
+        mapping<sample_name, list<string>> errors;
```

Comment:
I made it `sample_name` here because they're not guaranteed to be saved samples in system yet. The main utility of this function right now would be to verify the samples before they're saved in the sample_uploader so we can address all the errors upfront.